test/unit/test.py

```diff
@@ -288,37 +288,31 @@ def test_spec_or_default_1(self):
 
     def test_spec_or_default_2(self):
         test_spec = {
-            "existing_key": "true",
+            "existing_key": True,
             "existing_key_2": True,
-            "existing_key_3": "false",
+            "existing_key_3": False,
```

Comment:
We only need tests for ``True`` and ``False`` here. That means it would be enough to have two ``existing_keys``

As you can see below starting in line 307, the tests for the all keys do exactly the same thing